### ISSUE TITLE
Marginal standard deviation parameter σ ignored when calculating covariance matrix

### ISSUE THREAD

**Issue by matt-graham** [PieterjanRobbe/GaussianRandomFields.jl#45]
The definitions of `apply` for the covariance functions which have a marginal standard deviation parameter σ all seem to ignore this value resulting in covariance matrices computed [as described in the documentation](https://pieterjanrobbe.github.io/GaussianRandomFields.jl/stable/API/#GaussianRandomFields.apply) not being scaled correctly for example
```Julia
using GaussianRandomFields
exponential_covariance = CovarianceFunction(1, Exponential(1, σ=2.0))
pts = range(0, stop=1, length=11)
apply(exponential_covariance, pts)
```
gives output
```
11×11 LinearAlgebra.Symmetric{Float64, Matrix{Float64}}:
 1.0       0.904837  0.818731  0.740818  0.67032   0.606531  0.548812  0.496585  0.449329  0.40657   0.367879
 0.904837  1.0       0.904837  0.818731  0.740818  0.67032   0.606531  0.548812  0.496585  0.449329  0.40657
 0.818731  0.904837  1.0       0.904837  0.818731  0.740818  0.67032   0.606531  0.548812  0.496585  0.449329
 0.740818  0.818731  0.904837  1.0       0.904837  0.818731  0.740818  0.67032   0.606531  0.548812  0.496585
 0.67032   0.740818  0.818731  0.904837  1.0       0.904837  0.818731  0.740818  0.67032   0.606531  0.548812
 0.606531  0.67032   0.740818  0.818731  0.904837  1.0       0.904837  0.818731  0.740818  0.67032   0.606531
 0.548812  0.606531  0.67032   0.740818  0.818731  0.904837  1.0       0.904837  0.818731  0.740818  0.67032
 0.496585  0.548812  0.606531  0.67032   0.740818  0.818731  0.904837  1.0       0.904837  0.818731  0.740818
 0.449329  0.496585  0.548812  0.606531  0.67032   0.740818  0.818731  0.904837  1.0       0.904837  0.818731
 0.40657   0.449329  0.496585  0.548812  0.606531  0.67032   0.740818  0.818731  0.904837  1.0       0.904837
 0.367879  0.40657   0.449329  0.496585  0.548812  0.606531  0.67032   0.740818  0.818731  0.904837  1.0
```
while I think we would expect the diagonal values to have value 4.0 (and we get the exact same output if `σ = 1.0`). The σ values are used to scale the realizations generated using `sample` however. For example the following
```Julia
using Random
using Statistics
using GaussianRandomFields
exponential_covariance = CovarianceFunction(1, Exponential(1, σ=2.0))
pts = range(0, stop=1, length=11)
rng = MersenneTwister(7858353175566367368)
num_sample = 100000
cov([sample(GaussianRandomField(exponential_covariance, Cholesky(), pts), xi=randn(rng, size(pts))) for i in 1:num_sample])
```
gives the output
```
11×11 Matrix{Float64}:
 3.98961  3.59929  3.25215  2.93646  2.6572   2.40484  2.17514  1.96722  1.77992  1.62033  1.46528
 3.59929  3.97168  3.59148  3.24851  2.94032  2.65995  2.40696  2.17798  1.97328  1.79417  1.6239
 3.25215  3.59148  3.97137  3.59101  3.24629  2.93735  2.65832  2.40098  2.17673  1.97871  1.79247
 2.93646  3.24851  3.59101  3.96765  3.58573  3.24575  2.93926  2.65794  2.40693  2.18248  1.97588
 2.6572   2.94032  3.24629  3.58573  3.96014  3.58788  3.25225  2.94236  2.6654   2.41519  2.18939
 2.40484  2.65995  2.93735  3.24575  3.58788  3.97302  3.59644  3.25272  2.94536  2.66831  2.41426
 2.17514  2.40696  2.65832  2.93926  3.25225  3.59644  3.98382  3.59941  3.25724  2.94766  2.66522
 1.96722  2.17798  2.40098  2.65794  2.94236  3.25272  3.59941  3.96891  3.59195  3.24983  2.9401
 1.77992  1.97328  2.17673  2.40693  2.6654   2.94536  3.25724  3.59195  3.97639  3.60242  3.2569
 1.62033  1.79417  1.97871  2.18248  2.41519  2.66831  2.94766  3.24983  3.60242  3.98627  3.60501
 1.46528  1.6239   1.79247  1.97588  2.18939  2.41426  2.66522  2.9401   3.2569   3.60501  3.98925
```
which is within the expected sampling error of the output of
```Julia
using GaussianRandomFields
exponential_covariance = CovarianceFunction(1, Exponential(1, σ=2.0))
pts = range(0, stop=1, length=11)
apply(exponential_covariance, pts) * exponential_covariance.cov.σ^2
```
```
11×11 LinearAlgebra.Symmetric{Float64, Matrix{Float64}}:
 4.0      3.61935  3.27492  2.96327  2.68128  2.42612  2.19525  1.98634  1.79732  1.62628  1.47152
 3.61935  4.0      3.61935  3.27492  2.96327  2.68128  2.42612  2.19525  1.98634  1.79732  1.62628
 3.27492  3.61935  4.0      3.61935  3.27492  2.96327  2.68128  2.42612  2.19525  1.98634  1.79732
 2.96327  3.27492  3.61935  4.0      3.61935  3.27492  2.96327  2.68128  2.42612  2.19525  1.98634
 2.68128  2.96327  3.27492  3.61935  4.0      3.61935  3.27492  2.96327  2.68128  2.42612  2.19525
 2.42612  2.68128  2.96327  3.27492  3.61935  4.0      3.61935  3.27492  2.96327  2.68128  2.42612
 2.19525  2.42612  2.68128  2.96327  3.27492  3.61935  4.0      3.61935  3.27492  2.96327  2.68128
 1.98634  2.19525  2.42612  2.68128  2.96327  3.27492  3.61935  4.0      3.61935  3.27492  2.96327
 1.79732  1.98634  2.19525  2.42612  2.68128  2.96327  3.27492  3.61935  4.0      3.61935  3.27492
 1.62628  1.79732  1.98634  2.19525  2.42612  2.68128  2.96327  3.27492  3.61935  4.0      3.61935
 1.47152  1.62628  1.79732  1.98634  2.19525  2.42612  2.68128  2.96327  3.27492  3.61935  4.0
```

As a related issue in the mathematical definitions in the docstrings of the affected covariance functions, the covariances are all scaled by σ rather than σ² - as σ is described as a marginal standard deviation parameter, the (co)variances I think would be expected to scale with the square of σ rather than σ itself.

**Comment by msainsburydale** [comment]
This is a great package and it would be good to fix this small issue. Would this simply be a matter of adding a line of code at the end of `apply()` that multiples the output by `cov.σ^2`? I also agree that the docstrings should show the covariance functions being scaled by $\sigma^2$ rather than $\sigma$.

**Comment by PieterjanRobbe** [comment]
Thanks for pointing this out! Should be fixed by 1dc69279 now. I'll close the issue for now.
